src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -251,6 +311,35 @@ private async Task<bool> LoadOrReloadProjectAsync(ProjectToLoad projectToLoad, T
             return false;
         }
 
+        async Task<(LoadedProject, bool alreadyExists)> GetOrCreateProjectTargetAsync(ImmutableArray<LoadedProject> previousProjectTargets, ProjectFileInfo loadedProjectInfo)
+        {
+            var existingProject = previousProjectTargets.FirstOrDefault(p => p.GetTargetFramework() == loadedProjectInfo.TargetFramework);
+            if (existingProject != null)
+            {
+                return (existingProject, alreadyExists: true);
+            }
+
+            var targetFramework = loadedProjectInfo.TargetFramework;
+            var projectSystemName = targetFramework is null ? projectPath : $"{projectPath} (${targetFramework})";
+
+            var projectCreationInfo = new ProjectSystemProjectCreationInfo
+            {
+                AssemblyName = projectSystemName,
+                FilePath = projectPath,
+                CompilationOutputAssemblyFilePath = loadedProjectInfo.IntermediateOutputFilePath,
+            };
+
+            var projectSystemProject = await ProjectFactory.CreateAndAddToWorkspaceAsync(
+                projectSystemName,
+                loadedProjectInfo.Language,
+                projectCreationInfo,
+                _projectSystemHostInfo);
+
+            var loadedProject = new LoadedProject(projectSystemProject, ProjectFactory.Workspace.Services.SolutionServices, _fileChangeWatcher, _targetFrameworkManager);
+            loadedProject.NeedsReload += (_, _) => _projectsToReload.AddWork(projectToLoad with { ReportTelemetry = false });
```

Comment:
I think this implementation is kicking a new design-time build on each edit to the `app.cs` file. In follow-up PR we should probably make sure that a new design time build is only kicked if directives change, or if files that may be part of the project are added/removed, etc. Not for changes to ordinary source code, I think.

Reply:
I think there is an existing bug that we're kicking off design time builds when you save a .cs file in a regular project too.

Reply:
Tracking in #78559. Feel free to link any related bugs.